pkg/build/nodeimage/build.go

```diff
@@ -46,26 +50,95 @@ func Build(options ...Option) error {
 		ctx.logger.Warnf("unsupported architecture %q", ctx.arch)
 	}
 
-	// locate sources if no kubernetes source was specified
-	if ctx.kubeRoot == "" {
-		kubeRoot, err := kube.FindSource()
+	if ctx.buildType == "" {
+		ctx.buildType = detectBuildType(ctx.kubeParam)
+		if ctx.buildType != "" {
+			ctx.logger.V(0).Infof("Detected build type: %q", ctx.buildType)
+		}
+	}
+
```

Comment:
TODO consider to encapsulate this into `func getBuilder(buildType string) (builder, error)`

Reply:
will leave this for a follow up PR